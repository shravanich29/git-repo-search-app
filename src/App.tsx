import "./App.css";
import { MainComponent } from "./container/main";

function App() {
  return (
    <div className="App">
      <MainComponent
        loading={false}
        repoTableData={[]}
        getReposByUserName={() => {}}
      />
    </div>
  );
}

export default App;
