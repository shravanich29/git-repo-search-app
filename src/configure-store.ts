
import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import "regenerator-runtime/runtime";
import rootSaga  from "./container/main/redux/saga"
import { rootReducer } from "./container/main/redux/reducer"


const sagaMiddleware = createSagaMiddleware();

export const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);