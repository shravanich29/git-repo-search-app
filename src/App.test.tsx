import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { store } from "./configure-store";

test("renders learn react link", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
