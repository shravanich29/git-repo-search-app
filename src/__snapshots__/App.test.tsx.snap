// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="MainDiv"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="userName"
            placeholder="Enter username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-2"
              style="width: 0.01px;"
            >
              <span>
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained DeleteButton MuiButton-containedPrimary"
        id="searchRepoBtn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search Repository
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        aria-colcount="6"
        aria-multiselectable="false"
        aria-rowcount="0"
        class="MuiDataGrid-root MuiDataGrid-root"
        role="grid"
      >
        <div>
          <div />
        </div>
        <div
          class="MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-overlay"
            style="top: 56px;"
          >
            No rows
          </div>
          <div
            class="MuiDataGrid-columnsContainer"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-columnHeaderWrapper scroll"
              role="row"
              style="transform: translate3d(-0px, 0, 0); min-width: 300px;"
            >
              <div
                aria-colindex="1"
                class="MuiDataGrid-columnHeader--sortable MuiDataGrid-columnHeader"
                data-field="owner"
                role="columnheader"
                style="width: 50px; min-width: 50px; max-width: 50px;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle"
                      title=""
                    >
                      Owner
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiDataGrid-sortIcon MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-controls="mui-15273"
                      aria-haspopup="true"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      id="mui-40899"
                      tabindex="-1"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                class="MuiDataGrid-columnHeader--sortable MuiDataGrid-columnHeader"
                data-field="repoName"
                role="columnheader"
                style="width: 50px; min-width: 50px; max-width: 50px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle"
                      title=""
                    >
                      Repository Name
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiDataGrid-sortIcon MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-controls="mui-76260"
                      aria-haspopup="true"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      id="mui-98161"
                      tabindex="-1"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                class="MuiDataGrid-columnHeader--sortable MuiDataGrid-columnHeader"
                data-field="desc"
                role="columnheader"
                style="width: 50px; min-width: 50px; max-width: 50px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle"
                      title=""
                    >
                      Description
                    </div>
                    <div
                      class="MuiDataGrid-iconButtonContainer"
                    >
                      <button
                        aria-label="Sort"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        tabindex="-1"
                        title="Sort"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiDataGrid-sortIcon MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-controls="mui-90646"
                      aria-haspopup="true"
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      id="mui-48952"
                      tabindex="-1"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiDataGrid-cell"
                style="min-width: 150px; max-width: 150px; line-height: 55px; min-height: 56px; max-height: 56px;"
              />
            </div>
          </div>
          <div
            style="overflow: visible; height: 0px; width: 0px;"
          >
            <div
              class="MuiDataGrid-windowContainer"
              style="width: 0px; height: 0px;"
            >
              <div
                class="MuiDataGrid-window"
                style="top: 56px; overflow-y: auto;"
              >
                <div
                  class="MuiDataGrid-dataContainer"
                  style="min-width: 300px; min-height: 1px;"
                >
                  <div
                    class="MuiDataGrid-viewport"
                    style="min-width: 0; max-width: 0; max-height: 0;"
                  >
                    <div
                      class="MuiDataGrid-renderingZone"
                      style="max-height: 1px; width: 300px; transform: translate3d(-0px, -0px, 0);"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
        <div>
          <div
            class="MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption makeStyles-caption-6 MuiTypography-body2 MuiTypography-colorInherit"
                  id="mui-828"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiTablePagination-input makeStyles-input-7 MuiTablePagination-selectRoot"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-828 mui-22011"
                    class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                    id="mui-22011"
                    role="button"
                    tabindex="0"
                  >
                    25
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="25"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTablePagination-caption makeStyles-caption-6 MuiTypography-body2 MuiTypography-colorInherit"
                >
                  0-0 of 0
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Previous page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Previous page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Next page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Next page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
